### PR TITLE
fix: mf-5154 set ens as name of wallet when create/import ones

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-5154
https://mask.atlassian.net/browse/MF-5184

### CODE REVIEW

**Comment by UncleBill** [comment]
@guanbinrui PTAL

File: packages/mask/background/services/wallet/services/wallet/index.ts [hunk 1]
```diff
@@ -308,6 +318,26 @@ export function createWalletFromPrivateKey(name: string, privateKey: string) {
     return addWalletFromPrivateKey(ImportSource.LocalGenerated, name, privateKey)
 }
 
+export async function getWalletFromPrivateKey(privateKey: string) {
```
Comment (guanbinrui) [inline packages/mask/background/services/wallet/services/wallet/index.ts]:
```suggestion
export async function generateAddressFromPrivateKey(privateKey: string) {
```

File: packages/mask/background/services/wallet/services/wallet/index.ts [hunk 2]
```diff
@@ -320,6 +350,27 @@ export function recoverWalletFromPrivateKey(name: string, privateKey: string) {
     return addWalletFromPrivateKey(ImportSource.UserProvided, name, privateKey)
 }
 
+export async function getWalletFromKeyStoreJSON(json: string, jsonPassword: string) {
```
Comment (guanbinrui) [inline packages/mask/background/services/wallet/services/wallet/index.ts]:
```suggestion
export async function generateAddressFromKeyStoreJSON(json: string, jsonPassword: string) {
```

File: packages/mask/background/services/wallet/services/wallet/index.ts [hunk 3]
```diff
@@ -158,6 +158,58 @@ export async function deriveWallet(name: string) {
...
+    while (true) {
+        derivedTimes += 1
+
+        // protect from endless looping
+        if (derivedTimes >= MAX_DERIVE_COUNT) {
+            await database.updateWallet(primaryWallet.address, {
+                latestDerivationPath,
+            })
+            throw new Error('Exceed the max derivation times.')
+        }
```
Comment (guanbinrui) [inline packages/mask/background/services/wallet/services/wallet/index.ts]:
```suggestion
        if (derivedTimes >= MAX_DERIVE_COUNT) throw new Error('Exceed the max derivation times.')
```

File: packages/mask/background/services/wallet/services/wallet/index.ts [hunk 4]
```diff
@@ -308,6 +360,26 @@ export function createWalletFromPrivateKey(name: string, privateKey: string) {
     return addWalletFromPrivateKey(ImportSource.LocalGenerated, name, privateKey)
 }
 
+export async function generateWalletFromPrivateKey(privateKey: string) {
```
Comment (guanbinrui) [inline packages/mask/background/services/wallet/services/wallet/index.ts]: Align `generateAddressFromMnemonicWords`:

```suggestion
export async function generateAddressFromPrivateKey(privateKey: string) {
```

File: packages/mask/background/services/wallet/services/wallet/index.ts [hunk 5]
```diff
@@ -320,6 +392,27 @@ export function recoverWalletFromPrivateKey(name: string, privateKey: string) {
     return addWalletFromPrivateKey(ImportSource.UserProvided, name, privateKey)
 }
 
+export async function generateWalletFromKeyStoreJSON(json: string, jsonPassword: string) {
```
Comment (guanbinrui) [inline packages/mask/background/services/wallet/services/wallet/index.ts]: Align `generateAddressFromMnemonicWords`:

```suggestion
export async function generateAddressFromKeyStoreJSON(json: string, jsonPassword: string) {
```

**Pull request merged**